lib/tasks/update-package-json.test.js

```diff
@@ -0,0 +1,100 @@
+import { describe, it, beforeEach, expect, vi } from 'vitest';
+
+import updatePackageJson from './update-package-json';
+
+let files;
+
+vi.mock('node:fs/promises', () => {
+  return {
+    readFile: (filename) => {
+      return files[filename];
+    },
+    writeFile: (_, content) => {
+      files = JSON.parse(content);
+    },
+  };
+});
+
+describe('updatePackageJson() function', () => {
+  beforeEach(() => {
+    files = {};
+  });
+
+  it('wont error with an empty package.json', async () => {
+    files = {
+      'package.json': '{}',
+    };
+
+    await updatePackageJson();
+  });
+
+  it('updates the scripts', async () => {
+    files = {
+      'package.json': JSON.stringify({
+        scripts: {
+          build: 'ember build --environment=production',
+          lint: 'concurrently "pnpm:lint:*(!fix)" --names "lint:" --prefixColors auto',
+          'lint:fix':
+            'concurrently "pnpm:lint:*:fix" --names "fix:" --prefixColors auto',
+          'lint:hbs': 'ember-template-lint .',
+          'lint:hbs:fix': 'ember-template-lint . --fix',
+          'lint:js': 'eslint . --cache',
+          'lint:js:fix': 'eslint . --fix',
+          start: 'ember serve',
+          test: 'concurrently "pnpm:lint" "pnpm:test:*" --names "lint,test:" --prefixColors auto',
+          'test:ember': 'ember test',
+        },
+      }),
+    };
+
+    await updatePackageJson();
+    expect(files).toMatchSnapshot();
```

Comment:
so... the splitting of these tests is good for sure, it's nice to have verification for each test like this... but the problem is that a lot of the snapshots are the same so there is a lot of redundancy 

I imagined that you would do something like: 

```suggestion
    expect(JSON.parse(files['package.json').scripts).toMatchInlineSnapshot();
```

or use deepEqual for an object comparison

This way each test would be testing only the thing that it is stating that it is testing and it would be very clear in the test file the thing that it is comparing against without having to compare across 2 files.

Does all that make sense?